lib/cli/ui/spinner/spin_group.rb

```diff
@@ -143,6 +143,11 @@ def update_title(new_title)
             end
           end
 
+          sig { void }
+          def interrupt
+            @thread.raise(Interrupt)
```

Comment:
What happens when you send this to a thread that's already done? Should we only send this to running threads?

Reply:
good question, tried raising an exception on a dead thread and it seems to be a no-op, it does not trigger the thread's rescue